sdk/core/core/inc/az_span.h

```diff
@@ -55,7 +55,7 @@ AZ_NODISCARD AZ_INLINE int32_t az_span_capacity(az_span span) { return span._int
  * @brief Creates an empty span
  *
  */
-AZ_NODISCARD AZ_INLINE az_span az_span_null() { return (az_span){ 0 }; }
+#define AZ_SPAN_NULL (az_span){ 0 }
```

Comment:
Since az_span is defined in the same file, I'd again, define AZ_SPAN_NULL as

```
#include <stddef.h>
#define AZ_SPAN_NULL (az_span){ .ptr = NULL, .length = 0, .capacity = 0 }
```

because should you use AZ_SPAN_NULL when `_az_cfg.h` is not included (or, not between the `_az_span_prefix.h` and `_suffix.h`), GCC and Clang are going to produce warnings (which they shouldn't, but it's how the things currently are).

In any case, the final decision is up to you, if you considered that, and still think we should keep `{ 0 }` I am fine with the change either way. What do you need to complete this change? If you don't have access to the "Complete Pull Request" button, let me know, I can complete it.

Reply:
updated to use this per-member init. I can't merge so if you're good with it feel free to merge when appropriate.

Reply:
I just realized recently that you can't merge, - will do once CI/CD passes!
BTW, it looks like you need to update the branch.

Reply:
@danewalton I just realized recently that you can't merge, - will do once CI/CD passes!
BTW, it looks like you need to update the branch.

Reply:
yup currently fixing merge conflicts. will push an update